src/SignalR/clients/csharp/Client/test/UnitTests/CallbackRegistrationGeneratorTests.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Client.Tests
+{
+    public class CallbackRegistrationGeneratorTests
+    {
+        public interface IMyClient
+        {
+            void NoArg();
+            void SingleArg(int a);
+            void ManyArgs(int a, float b);
+            Task ReturnTask();
+        }
+
+        private class MyClient : IMyClient
+        {
+            public int CallsOfNoArg;
+            public void NoArg()
+            {
+                CallsOfNoArg += 1;
+            }
+
+            public List<int> CallsOfSingleArg = new();
+            public void SingleArg(int a)
+            {
+                CallsOfSingleArg.Add(a);
+            }
+
+            public List<(int, float)> CallsOfManyArgs = new();
+            public void ManyArgs(int a, float b)
+            {
+                CallsOfManyArgs.Add((a, b));
+            }
+
+            public Task ReturnTask()
```

Comment:
Looks like some tests need to be updated now that this is added

Reply:
Added more tests overall. I wasn't sure initially whether to add tests for each interface method (since many issues will result in compile errors of `src/SignalR/clients/csharp/Client/test` anyhow) however all the example hub proxy and callback provider methods should now be covered.